src/util/sock.cpp

```diff
@@ -191,8 +192,17 @@ std::string Sock::RecvUntilTerminator(uint8_t terminator,
 
     for (;;) {
         char buf[512];
+        size_t peek_bytes{sizeof(buf)};
 
-        const ssize_t peek_ret{Recv(buf, sizeof(buf), MSG_PEEK)};
+        if (max_data.has_value()) {
+            if (data.size() >= max_data.value()) {
+                throw std::runtime_error(
+                    strprintf("Received too many bytes without a terminator (%u)", data.size()));
+            }
+            peek_bytes = std::min(sizeof(buf), max_data.value() - data.size());
+        }
```

Comment:
The dereference operator `operator*()` does not check if this optional contains a value, which may be more efficient than `value()` (https://en.cppreference.com/w/cpp/utility/optional/value). As `has_value()` is already checked, can also do (maybe not worth it but as a note):

```diff
         if (max_data.has_value()) {
-            if (data.size() >= max_data.value()) {
+            const size_t bytes{data.size()};
+            if (bytes >= *max_data) {
                 throw std::runtime_error(
-                    strprintf("Received too many bytes without a terminator (%u)", data.size()));
+                    strprintf("Received too many bytes without a terminator (%u)", bytes));
             }
-            peek_bytes = std::min(sizeof(buf), max_data.value() - data.size());
+            peek_bytes = std::min(sizeof(buf), *max_data - bytes);
         }
 ```

Reply:
Yeah, there is more than one way to write this. I chose the current one because I think the performance of `operator*` is negligible in this case and I find `.value()` more readable. Also
```cpp
if (max_data.has_value()) {
```
can be written as
```cpp
if (max_data) {
```
but I don't like those "hidden" casts and somehow prefer the former. The latter raises the question: "given that `max_data` represents a number, did the programmer intend to check for non-zero value instead of whether the optional has value or not?"

I also prefer explicit comparisons (avoid implicit cast to boolean):
```cpp
if (foo != 0) { // makes it obvious that foo is a number
if (foo != nullptr) { // makes it obvious that foo is a pointer
if (foo) { // use this only if foo is a boolean
```

Reply:
I agree. It was more of a note to expose options.